### ISSUE TITLE
[Accessibility] Narrator/Inspect hang eventually when the "ComboBox" to be expanded contains duplicate items

### ISSUE THREAD

**Issue by Jessie-Zhang01** [dotnet/winforms#3590]
* .NET Core Version: 
.Net 5.0.100-rc.1.20365.11 from 5.0 master branch

* Have you experienced this same bug with .NET Framework?: 
Yes, it has a VSO bug: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1157357

**More Info:**

1. This issue cannot reproduce when set DropDownStyle property of ComboBox as 'Simple'.
2. This issue can reproduce when set DropDownStyle property of ComboBox as 'DropDown/DropDownList'.
3. This issue also can reproduce on .Net 3.1.

**Minimal repro:**

1. Open the attached ‘WindowsFormsApp4’ application in VS, then build and run it.
2. Open Narrator or Inspect, then focus its cursor to the ComboBox control.
3. Expand the ComboBox drop-down list.
[WindowsFormsApp4.zip](https://github.com/dotnet/winforms/files/4929733/WindowsFormsApp4.zip)

**Expected behavior:**
Narrator or Inspect's cursor can be focus on the ComboBox‘s items and works fine.

**Actual behavior:**
Narrator/Inspect will hang.
![Issue-1](https://user-images.githubusercontent.com/45864649/87739556-bb908a80-c812-11ea-91a6-72b5b3ef13de.gif)

**Comment by AaronRobinsonMSFT** [comment]
@Jessie-Zhang01 and @TanayParikh I think this is the same issue as https://github.com/dotnet/winforms/pull/3524. I will continue to add notes there.

**Comment by Jessie-Zhang01** [comment]
Verified the issue with latest NET SDK 6.0.100-alpha.1.21060.2, this issue is fixed. Narrator or Inspect's cursor can be focus on the ComboBox‘s items and works fine.
![step](https://user-images.githubusercontent.com/45864649/104154570-36340e80-5420-11eb-9cc9-b30c7d4459ca.gif)
